### DISCUSSION TITLE
Add Unique ID field to product

### DISCUSSION THREAD

**Discussion by nathanss** [woocommerce/woocommerce#47519] [General]
We've been studying adding a new field (provisory name: Unique ID) to the product. The initial idea, described in https://github.com/woocommerce/woocommerce/issues/39577, is that this field can represent multiple unique identifiers, like GTIN, UPC, EAN, or ISBN. It also makes sense to add it to the Inventory tab of the legacy editor, if it's useful. This is a common demand and some extensions already implement similar behavior. We want to gather feedback before we decide how we should approach this. 

Some points to discuss:

* Should the field have a 'uniqueness' validation, similar to SKU?
* Should the field have additional validations?
* How should it be represented in the storefront? (I guess that it should be similar to SKU)
* Should this field appear in other application parts, like the 'All products' table?

I've written a PoC that adds the field to the product and the new product editor. [You can check it out here.](https://github.com/woocommerce/woocommerce/pull/47364)

**Extended deadline: June 17**

**Comment by barryhughes** [comment]
> This is a common demand and some extensions already implement similar behavior. 

My initial thought was, this does indeed seem like extension territory ... but if there is sufficient demand and adding this generally reduces friction, it could be a nice win.

<div align="center"><img width="700" alt="Screenshot 2024-05-15 at 10 43 37" src="https://github.com/woocommerce/woocommerce/assets/3594411/0c33b7db-7405-476a-a798-413f7353b1ac"></div>

I'm also curious about how it would work on the frontend—would there be a corresponding setting to specify what the field actually *is* (ie, a setting to designate it as the ISBN field rather than a EAN field)?

Fwiw, reminder here that we actually already support custom product fields (or, inherit them from WordPress) which can be configured on a post-by-post basis, and perhaps this could be complimented by new functionality that allows them to be defined globally—could that be the better/more flexible solution?

**Comment by pmcpinto** [comment]
Expanding on Nathan's points, we're evaluating adding this as a Core option because it holds significant potential benefits for a large number of merchants. These include:

- Better inventory management and tracking through standardized product identification methods
- Enhanced data accuracy
- Improved searchability and SEO in search engines and marketplaces
- And more

**Comment by fmgconsultants** [comment]
I think this is an excellent idea - we are always having to add custom fields into our Woo single product page templates for this very purpose.  

I don't see this as an extension but a core feature.  There isn't hardly a site out there these days that doesn't need the ability to add GTIN, ISBN, UPC, etc. to they can take advantage of the Google Merchant Center or Amazon Listings.

Our clients are industrial manufacturers (OEMs) and they use a variety of ERPs - having this as core, would make it a lot easier to tie into / map to their ERP fields for inventory management.

**Comment by betayazilim** [comment]
google merchant requires model name, EAN, MPN and brand.
Which also required many other platforms like price comparison websites.
We can't define a product without these 4 attributes. and yes, unfortunately we use attributes for all these 4.
